jira_scraper.py

```diff
@@ -58,6 +57,59 @@ def get_jira_data(
 
     return data["issues"]
 
+def assemble_jira_query(jira_projects: list = DEFAULT_JIRA_PROJECTS) -> str:
+    projects = " OR ".join([f"project={e}" for e in jira_projects])
+
+    # Define the terms to search for in descriptions and comments
+    search_terms = [
+        "root cause",
+        "RCA",
+        "failure description",
+        "impacted component",
+        "resolved"
+    ]
+
+    # Create the description and comment search part
+    desc_comment_search = " OR ".join([
+        f'description ~ "{term}"' for term in search_terms
+    ] + [
+        f'comment ~ "{term}"' for term in search_terms
+    ])
+
+    pr_search = " OR ".join([
+        'description ~ "github.com"',
+        'description ~ "review.opendev.org"',
+        'description ~ "gitlab.*com"', # for GitLab domains
+        'comment ~ "github.com"',
+        'comment ~ "review.opendev.org"',
+        'comment ~ "gitlab.*com"',
+        'description ~ "pull request"',
+        'comment ~ "pull request"',
+        'description ~ "merged"',
+        'comment ~ "merged"'
+    ])
+
```

Comment:
Can we share the difference in results we get. I don't think a description would normally have PR/patch details.

Reply:
updating the MR.